tasks/add-unique-granuleID/src/types.ts

```diff
@@ -0,0 +1,23 @@
+// Either collectionId OR (datatype AND version) must be present
+export type GranuleInput =
+  | (BaseGranule & { collectionId: string; dataType: never; version: never })
```

Comment:
Is this syntax right?

Reply:
It is - you can add an extra '|' for visual clarity.   It's a formatting convention, it could be removed and winds up being the same.  

Prettier also recommends it (although that's not where this came from in this instance)

It's in the official prettier docs https://github.com/prettier/prettier/commit/d062961ccfec6ccf77e49fa2fc29b01b92894161

as well as https://github.com/microsoft/TypeScript/issues/12071